java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/BinaryConsumer.java

```diff
@@ -45,6 +45,9 @@
    * Instantiate a BinaryConsumer.
    */
   public BinaryConsumer(VarBinaryVector vector, int index) {
+    if (vector != null) {
+      vector.allocateNewSafe();
```

Comment:
can you clarify why this is necessary?

Reply:
Since we remove JdbcToArrowUtils#allocateVectors, the vector is initialized with capacity 0, and when we invoke vector.getOffsetBuffer.get(index *4) it will throw Exception.
And the vector passed in may be null from iterator API, so add a null check also.